### PR TITLE
feat(bulk-model-sync-gradle): add continueOnError option

### PR DESCRIPTION
**Pull request by languitar**
With the new option, the sync can ignore errors encountered during syncing instead of immediately terminating. This is some sort of graceful degradation at the cost of potentially resulting in an inconsistent sync.

I'll add the missing implementation for the direction towards MPS after #312 is merged to avoid unnecessary conflicts.

### CODE REVIEW

File: bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/config/ModelSyncGradleSettings.kt
Comment (mhuster23) [inline bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/config/ModelSyncGradleSettings.kt:46]: I wonder, if we should expose this only as a method like `enableContinueOnError()` for consistency as sync direction only used methods for its settings up until now.

File: docs/global/modules/core/pages/reference/component-bulk-model-sync-gradle.adoc
Comment (mhuster23) [inline docs/global/modules/core/pages/reference/component-bulk-model-sync-gradle.adoc:82]: The table is broken because line 77 hasn't been removed.
Also the content does not fit the table header since it is not a method.

File: bulk-model-sync-lib/src/commonMain/kotlin/org/modelix/model/sync/bulk/ModelImporter.kt
Comment (mhuster23) [inline bulk-model-sync-lib/src/commonMain/kotlin/org/modelix/model/sync/bulk/ModelImporter.kt:50]:
```suggestion
    private fun doAndPotentiallyContinueOnErrors(block: () -> Unit) {
```
This should make it clearer. Still the method name is a bit long but I have a hard time coming up with a more concise name.

File: bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt
Comment (mhuster23) [inline bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt:89]: `continueOnError` only affects the import into the model server this way. For the import into MPS the solution must be changed as well.

**Review by mhuster23 — CHANGES_REQUESTED** [review]

File: bulk-model-sync-lib/src/commonMain/kotlin/org/modelix/model/sync/bulk/ModelImporter.kt
Comment (languitar) [reply bulk-model-sync-lib/src/commonMain/kotlin/org/modelix/model/sync/bulk/ModelImporter.kt:50]: The missing "on" was an oversight.

File: bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/config/ModelSyncGradleSettings.kt
Comment (languitar) [reply bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/config/ModelSyncGradleSettings.kt:46]: I had that in my Gradle script and it looked a bit weird to me for just a single boolean.

File: bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt
Comment (mhuster23) [reply bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt:89]: If you want it only for this direction, I suggest moving the `continueOnError` property to the `ServerTarget` so it's clear for the user.
For the other direction you could also add `continueOnError` for `LocalSource` and then pass the value via a system property in the ant script similar to includedModules. But maybe you should wait for #312. Since those changes will make it easier.

File: docs/global/modules/core/pages/reference/component-bulk-model-sync-gradle.adoc
Comment (languitar) [reply docs/global/modules/core/pages/reference/component-bulk-model-sync-gradle.adoc]: fixed

File: bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt
Comment (languitar) [reply bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt:89]: I can rebase my branch on that one and we wait for it to be merged.

Comment (languitar) [reply bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt:89]: I think I am stuck here. I'd want to implement that also for the direction into MPS. However, due to [#312 being on an outdated state of the sync](https://github.com/modelix/modelix.core/pull/312/files#r1390940876), I can easily built that feature in the current PR. I could log a warning at the moment and implement that once #312 is done.

File: bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/config/ModelSyncGradleSettings.kt
Comment (languitar) [reply bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/config/ModelSyncGradleSettings.kt]: Updated to that pattern

File: bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt
Comment (languitar) [reply bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ImportIntoModelServer.kt:89]: Is that ok with the message for the moment to unblock this?

File: bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/config/ModelSyncGradleSettings.kt
Comment (mhuster23) [reply bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/config/ModelSyncGradleSettings.kt]: The name in the docs isn't updated. We could also get rid of the parameter as `enableContinueOnError(false)` wouldn't do anything.

Comment (languitar) [reply bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/config/ModelSyncGradleSettings.kt]: without the flag, that would be a one-way street. I added the flag so that one is also able to disable this, too.

I'll update the docs

**Review by mhuster23 — APPROVED** [review]

**Pull request merged**
